File: fizz_buzz/fizzbuzz.py
#====================================================================
#=                            FIZZ BUZZ                             =                    
#====================================================================

# Problem: Write a program that outputs the string representation of numbers from 1 to n. 
#          For each multiple of 3, print 'Fizz' instead of the number. 
#          For each multiple of 5, print 'Buzz' instead of the number. 
#          For numbers which are multiples of both 3 and 5, print 'FizzBuzz' instead of the number.
#
#  Time Complexity: O(n)
# Space Complexity: O(n)

class Solution:
    def fizzy(self, num):
        list = []
        for i in range(1, num + 1):
            if i % 15 == 0: list.append('FizzBuzz')
            elif i % 3 == 0: list.append('Fizz')
            elif i % 5 == 0: list.append('Buzz')
            else: list.append(str(i))
        return '\n'.join(list)

def main():
    s = Solution()
    print(s.fizzy(100))

if __name__ == "__main__":
    main()

File: fizz_buzz/fizzbuzz.cpp
//====================================================================
//=                            FIZZ BUZZ                             =                    
//====================================================================

// Problem: Write a program that outputs the string representation of numbers from 1 to n.
//          For each multiple of 3, print 'Fizz' instead of the number. 
//          For each multiple of 5, print 'Buzz' instead of the number. 
//          For numbers which are multiples of both 3 and 5, print 'FizzBuzz' instead of the number.
//
//  Time Complexity: O(n)
// Space Complexity: O(n)

#include <iostream>

class Solution {
    public:
        void fizzy() {
            for (int i = 1; i <= 100; i++) {
                if (i % 15 == 0) std::cout << "FizzBuzz\n";
                else if (i % 3 == 0) std::cout << "Fizz\n";
                else if (i % 5 == 0) std::cout << "Buzz\n";
                else std::cout << i << "\n";
            }    
        }
};

int main() {
    Solution s;
    s.fizzy();
}